src/scenes/Partnerships/Edit/EditPartnership.tsx

```diff
@@ -0,0 +1,117 @@
+import { Box, Button } from '@material-ui/core';
+import React, { FC } from 'react';
+import { Except } from 'type-fest';
+import {
+  displayPartnershipStatus,
+  GQLOperations,
+  PartnershipStatuses,
+  PartnershipType,
+  UpdatePartnershipInput,
+} from '../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../../../components/form';
+import {
+  EditPartnershipFragment,
+  useDeletePartnershipMutation,
+  useUpdatePartnershipMutation,
+} from './EditPartnership.generated';
+
+type EditPartnershipProps = Except<
+  DialogFormProps<UpdatePartnershipInput>,
+  'onSubmit' | 'initialValues'
+> & {
+  partnership: EditPartnershipFragment;
+};
+
+export const EditPartnership: FC<EditPartnershipProps> = ({
+  partnership,
+  ...props
+}) => {
+  const [updatePartnership] = useUpdatePartnershipMutation();
+  const [deletePartnership] = useDeletePartnershipMutation();
+
+  const onDeleteClick = () => {
+    const { id } = partnership;
+    deletePartnership({
+      variables: { input: id },
+      refetchQueries: [GQLOperations.Query.ProjectPartnerships],
+    });
+    //TODO: correctly handle closing the dialog.  props.onClose has a TS errors
+    // props.onClose && props.onClose('cancel');
+  };
+
+  const radioOptions = PartnershipStatuses.map((status) => (
+    <RadioOption label={displayPartnershipStatus(status)} value={status} />
+  ));
+
+  return (
+    <DialogForm<UpdatePartnershipInput>
+      //TODO: fix the form flashing back to initialValues
+      onSubmit={(input) => {
+        updatePartnership({
```

Comment:
You need to await the `updatePartnership` call. Does that fix the initial values flashing back?

Reply:
No that doesn't fix. here's the code I tried
```
 onSubmit={async (input) => {
        await updatePartnership({
          variables: { input },
          refetchQueries: [GQLOperations.Query.ProjectPartnerships],
        });
      }}
```

Reply:
Hmm ok. Let's solve it in a separate pass. This will probably be happening everywhere.